pkg/resourcecollector/resourcecollector.go

```diff
@@ -522,6 +523,37 @@ func (r *ResourceCollector) GetResources(
 	return allObjects, pvcsWithOwnerReference, nil
 }
 
+// gatherResourceInChunks() collects all the resources present per ns or resource types.
+// It does it in batch of 500 unless specified by user via a config-map. This function
+// primarily helps in avoiding server timeout error when the number of resource or size
+// of it large.
+func gatherResourceInChunks(dynamicClient dynamic.ResourceInterface, opts Options, selectors string) (*unstructured.UnstructuredList, error) {
+	var err error
+	fn := "gatherResourceInChunks"
+	objectsList := &unstructured.UnstructuredList{}
+	var temp *unstructured.UnstructuredList
+	listOps := metav1.ListOptions{
+		LabelSelector: selectors,
+		Limit:         opts.ResourceCountLimit,
+	}
+	for {
+		temp, err = dynamicClient.List(context.TODO(), listOps)
+		if err != nil {
+			logrus.Infof("%v: failed to list resources", fn)
```

Comment:
nit: should be errorf()

Reply:
I made it as an Warnf and anyway the outer function shows the final error anyway